k8s/upgrades/0.9.0-1.0.0/cstor_pool_upgrade.sh

```diff
@@ -0,0 +1,279 @@
+                          tr "[]" " ")
+    local spc_bd_list=""
+    local no_of_blockdevices=$(echo $spc_disk_list | wc -w)
+
+    ##Patch SPC only if it is manual provisioning
+    if [ $no_of_blockdevices != 0 ]; then
+        for disk_name in $spc_disk_list; do
+             spc_bd_list=$(make_spc_blockdevice_list "$spc_bd_list" "$disk_name")
+        done
+        sed "s|@blockdevice_list@|$spc_bd_list|g" spc-patch.tpl.json > spc-patch.json
+
+        kubectl patch spc $spc_name -p "$(cat spc-patch.json)" --type=merge
+        rc=$?
+        if [ $rc -ne 0 ]; then
+            echo "Failed to patch spc: $spc_name with block device information | Exit Code: $rc"
+            error_msg
+            rm spc-patch.json
+            exit 1
+        fi
+
+        rm spc-patch.json
+    fi
+}
+
+## make_csp_disk_list will return the jsonpath required to update the csp
+function make_csp_disk_list() {
+    local csp_disk_list=$1
+    local disk_device_id=$2
+    local disk_name=$3
+    if [ -z "$csp_disk_list" ]; then
+        echo "{\"deviceID\": \"$disk_device_id\",\"inUseByPool\": true,\"name\": \"$disk_name\"}"
+    else
+        echo "$csp_disk_list,{\"deviceID\": \"$disk_device_id\",\"inUseByPool\": true,\"name\": \"$disk_name\"}"
+    fi
+}
+
+########################################################################
+#                                                                      #
+#                          Starting point                              #
+#                                                                      #
+########################################################################
+if [ "$#" -ne 2 ]; then
+    usage
+fi
+
+spc=$1
+ns=$2
+declare -A csp_blockdevice_list
+
+### Get the deployment pods which are in not running state that are related to provided spc ###
+pending_pods=$(kubectl get po -n $ns \
+    -l app=cstor-pool,openebs.io/storage-pool-claim=$spc \
+    -o jsonpath='{.items[?(@.status.phase!="Running")].metadata.name}')
+rc=$?; if [ $rc -ne 0 ]; then echo "Failed to get pool pods related to spc: $spc | Exit Code: $rc"; error_msg; exit 1; fi
+
+
+## If any deployments pods are in not running state then exit the upgrade process ###
+if [ $(echo $pending_pods | wc -w) -ne 0 ]; then
+    echo "To continue with upgrade script make sure all the pool deployment pods corresponding to $spc must be in running state"
+    error_msg
+    exit 1
+fi
+
+patch_blockdevice_list_for_spc $spc
+
+### Get the csp list which are related to the given spc ###
+csp_list=$(get_csp_list $spc)
+
+#### Get required info from current csp and use the info while upgrading ####
+for csp_name in `echo $csp_list | tr ":" " "`; do
+    ## Check CSP version ##
+    version=$(verify_openebs_version "csp" $csp_name)
+    rc=$?
+    if [ $rc -ne 0 ]; then
+        exit 1
+    elif [ $version == $pool_upgrade_version ]; then
+        continue
+    fi
+
+    ## Get disk info from corresponding sp ##
+    sp_name=$(kubectl get sp \
```

Comment:
are we handling deletion of sp as part of upgrade?

Reply:
Yes deprecating sp should be part of upgrade